<?xml version="1.0"?>
<launch>
  
  <!-- Arguments -->
  <arg name="disturbance_link" default="manipulator::wrist_3_link"/>
  <arg name="reference_frame" default="world"/>
  <arg name="force_duration" default="100.0"/>

  <!-- Start the disturbance applier node -->
  <node name="disturbance_applier_node" 
        pkg="manipulator_disturbance_control" 
        type="disturbance_applier_node"
        output="screen">
    <param name="disturbance_link" value="$(arg disturbance_link)"/>
    <param name="reference_frame" value="$(arg reference_frame)"/>
    <param name="force_duration" value="$(arg force_duration)"/>
  </node>

  <!-- Optional: Launch RViz for visualization -->
  <node name="rviz" 
        pkg="rviz" 
        type="rviz" 
        args="-d $(find manipulator_disturbance_control)/config/manipulator_view.rviz"
        if="false"/>

</launch>
